src/keyDerivation_test.c

```diff
@@ -0,0 +1,141 @@
+#include "keyDerivation.h"
+#include "test_utils.h"
+#include "hex_utils.h"
+#include "stream.h"
+#include <string.h>
+#include "errors.h"
+
+#define ARRAYLEN(array) (sizeof(array) / sizeof(array[0]))
+
+void testPrivateKeyDerivation()
+{
+#define TESTCASE(path_, pathLength_, expected_) \
+	{ \
+		uint8_t chainCode[32]; \
+		cx_ecfp_256_extended_private_key_t privateKey; \
+		derivePrivateKey(path_, pathLength_, chainCode, &privateKey); \
+		stream_t s; \
+		stream_init(&s); \
```

Comment:
Looking at all our testing code I guess we should make `stream_initFromHexString` :-)

Reply:
Yep, next PR ? :factory:

Reply:
Great!